python/pyspark/sql/dataframe.py

```diff
@@ -722,8 +722,6 @@ def __getitem__(self, item):
         [Row(age=5, name=u'Bob')]
         """
         if isinstance(item, basestring):
-            if item not in self.columns:
```

Comment:
We already capture the AnalysisException. If it's already good enough, we can just remove this check.